node/builder.go

```diff
@@ -152,7 +153,7 @@ func libp2p() Option {
 
 		Override(new(lp2p.RawHost), lp2p.Host),
 		Override(new(host.Host), lp2p.RoutedHost),
-		Override(new(lp2p.BaseIpfsRouting), lp2p.DHTRouting(false)),
+		Override(new(lp2p.BaseIpfsRouting), lp2p.DHTRouting(dht.ModeAuto)),
```

Comment:
@whyrusleeping @Stebalien I'm not sure what the correct default mode for Lotus should be, since it depends on the DHTs expected usage.

In go-ipfs users are using the DHT with all sorts of strange network configurations and so we've adapted to protect the network with an "Auto" mode that switches between client and server modes based on whether the peer thinks it is publicly accessible. However, it may be that Lotus users are generally expected to be publicly accessible and to have figured out their own NAT traversal stories in which case using the "Server" mode isn't crazy.

Reply:
I'd start with Server for now. We can decide later whether or not to start in Auto. But if we _do_ start in Auto, we'll have to make sure to start the AutoNAT service and I'm not sure if lotus nodes want to do that.